cvat-ui/plugins/sam/src/ts/index.tsx

```diff
@@ -183,90 +214,119 @@ const samPlugin: SAMPlugin = {
+                                    const image = Array(height).fill(0);
+                                    for (let i = 0; i < image.length; i++) {
+                                        image[i] = Array(width).fill(0);
+                                    }
+
+                                    for (let i = 0; i < input.length; i++) {
+                                        const row = Math.floor(i / width);
+                                        const col = i % width;
+                                        image[row][col] = input[i] > 0 ? 255 : 0;
+                                    }
+
+                                    return image;
+                                }
+
+                                function onnxToImage(input: any, width: number, height: number): number[][] {
+                                    return toMatImage(input, width, height);
+                                }
+
+                                plugin.data.worker.postMessage({
+                                    action: WorkerAction.DECODE,
+                                    payload: feeds,
+                                });
+
+                                plugin.data.worker.onmessage = ((e) => {
+                                    if (e.data.action !== WorkerAction.DECODE) {
+                                        const error = 'Caught unexpected action response from worker: ' +
+                                                `${e.data.action}, while "${WorkerAction.DECODE}" was expected`;
+                                        reject(new Error(error));
+                                    }
+
+                                    if (!e.data.error) {
+                                        const {
+                                            masks, lowResMasks, xtl, ytl, xbr, ybr,
+                                        } = e.data.payload;
+                                        const imageData = onnxToImage(masks.data, masks.dims[3], masks.dims[2]);
+                                        plugin.data.lowResMasks.set(key, lowResMasks);
+
+                                        resolve({
+                                            mask: imageData,
+                                            bounds: [xtl, ytl, xbr, ybr],
+                                        });
+                                    } else {
+                                        reject(new Error(`Decoder error. ${e.data.error}`));
+                                    }
+                                });
+
+                                plugin.data.worker.onerror = ((error) => {
+                                    reject(error);
+                                });
+                            });
+                    });
                 },
             },
         },
     },
     data: {
+        initialized: false,
         core: null,
+        worker: new Worker(new URL('./inference.worker', import.meta.url)),
```

Comment:
Please, reduce number of embeddings and lowResMasks stored in RAM as it done for efficientvit